### ISSUE TITLE
Create and support a PPA

### ISSUE THREAD

**Issue by overheadhunter** [AdoptOpenJDK/openjdk-installer#201]
As suggested by @aahlenst in https://github.com/AdoptOpenJDK/openjdk-installer/issues/10#issuecomment-606524942, I open a new issue, because I think many downstream PPA maintainers would benefit of AdoptOpenJDK being published in a PPA.

### Rationale

I want to use AdoptOpenJDK as a build time dependency in my own PPA. However, one can only depend on packages within Ubuntu's main repo or on packages from different PPAs.

In order to publish packages in a PPA, you need to create a debian source file (`debuild -S`) and upload it together with the original source taball to the build system. The build itself will run "offline", i.e. we can not download additional packages at build time. Therefore it is not possible to depend on @rpardini's PPA, which just downloads the pre-built binaries.

### Alternatives

One could use Ubuntu's OpenJDK builds, but they [have some issues preventing me from using `jlink`](https://bugs.launchpad.net/ubuntu/+source/openjdk-14/+bug/1868699). Also, one can't use recent OpenJDK builds on older distros and have to live with their release cycle.

**Comment by overheadhunter** [comment]
@aahlenst
> Is it possible for us the upload the pre-built packages to a PPA?

No, you need to upload source packages. Only binaries built by the PPA build system get published.

**Comment by aahlenst** [comment]
This is going to be tough one. We have an extensive test suite that we run as part of the build process. It ensures that our packages work. This cannot be easily ported, would be needless duplication and I'm not sure the PPA system could handle it. Besides, we're not even using debuild to build our packages.

Do you have a contact at Ubuntu? Maybe we can work something out with them.

**Comment by overheadhunter** [comment]
> Do you have a contact at Ubuntu? Maybe we can work something out with them.

I think @doko42 is responsible for the openjdk builds on Ubuntu. Doesn't seem to be too active on GitHub, though. There are some further contact options [on his launchpad profile](https://launchpad.net/~doko).

**Comment by aahlenst** [comment]
So far, we haven't found the right people to talk to at Canonical.

At [Adoptium](https://projects.eclipse.org/projects/adoptium/governance), we won't be allowed to distribute builds that haven't gone through the JCK. The JCK is a proprietary Oracle product and access to it is highly restricted. Therefore, building over at Launchpad won't be possible. Even if Canonical entered into a JCK agreement with Oracle, I doubt that building at Launchpad would be possible.
